sdk/azidentity/chained_token_credential.go

```diff
@@ -14,15 +14,22 @@ import (
 
 // ChainedTokenCredentialOptions contains optional parameters for ChainedTokenCredential.
 type ChainedTokenCredentialOptions struct {
-	// placeholder for future options
+	// If true, it will not assume the first successful credential should be always used.
+	RetryAllSources bool
 }
 
 // ChainedTokenCredential is a chain of credentials that enables fallback behavior when a credential can't authenticate.
+// By default, this credential will assume that the first successful credential should be the only credential used on future requests.
+// When `retryAllSources` is true, it will always try to get a token with every credential available on the `sources` array.
```

Comment:
Since `retryAllSources` isn't exported, we don't need to publicly doc it (plus it's doc'ed in the options).